dig.go

```diff
@@ -377,7 +377,16 @@ func (c *Container) provide(ctor interface{}, opts provideOptions) error {
 			return err
 		}
 	}
-	c.dg.Add(n.paramList.DotNodes(), n.resultList.DotNodes())
+
+	constructor := &dot.Ctor{
+		Name:    n.location.Name,
+		Package: n.location.Package,
+		File:    n.location.File,
+		Line:    n.location.Line,
+		Params:  n.paramList.DotNodes(),
+		Results: n.resultList.DotNodes(),
+	}
+	c.dg.Ctors = append(c.dg.Ctors, constructor)
```

Comment:
An alternative would be to use the `dot.Graph.Add` method like before:

```
func (dg *Graph) Add(c *Ctor) {
  dg.Ctors = append(dg.Ctors, c)
}
```

```
c.dg.Add(&dot.Ctor{
  Name:    n.location.Name,
  Package: n.location.Package,
  File:    n.location.File,
  Line:    n.location.Line,
  Params:  n.paramList.DotNodes(),
  Results: n.resultList.DotNodes(),
})
```

But seeing as this is largely the same, I'll leave it up to you!